proxy/src/main/java/net/md_5/bungee/ServerConnection.java

```diff
@@ -25,8 +28,7 @@
     @Getter
     private final boolean forgeServer = false;
     @Getter
-    @Setter
-    private long sentPingId = -1;
+    private final Queue<KeepAliveData> keepAliveHistory = EvictingQueue.create(1024);
```

Comment:
EvictingQueue is presized with an array of this - 1024 seems a bit excessive.
Right now the existing code is essentially an EvictingQueue of size 1 and things are going (mostly ok), surely even eg 32 would be plenty.

Reply:
If a server wants to send a ping packet every tick, I feel the client should be given the chance to not reply for up to 30 seconds, the default timeout. That's at least 600, but I feel we should be generous given that it doesn't take much memory to store this data.